src/models/unit_tests/rule_test.rs

```diff
@@ -10,42 +10,56 @@ Copyright (c) 2022 Uber Technologies, Inc.
  express or implied. See the License for the specific language governing permissions and
  limitations under the License.
 */
-use crate::models::{default_configs::JAVA, language::PiranhaLanguage};
+use crate::{
+  constraint,
+  models::{default_configs::JAVA, language::PiranhaLanguage},
+  utilities::eq_without_whitespace,
+};
 use std::collections::HashSet;
 
 use super::InstantiatedRule;
 use {
-  super::Rule,
-  crate::models::{
-    constraint::Constraint, rule_store::RuleStore, source_code_unit::SourceCodeUnit,
-  },
+  crate::models::{rule_store::RuleStore, source_code_unit::SourceCodeUnit},
   std::collections::HashMap,
   std::path::PathBuf,
 };
 
 /// Tests whether a valid rule can be correctly instantiated given valid substitutions.
 #[test]
 fn test_rule_try_instantiate_positive() {
-  let holes = HashSet::from([String::from("variable_name")]);
-  let rule = Rule::new("test","(((assignment_expression left: (_) @a.lhs right: (_) @a.rhs) @abc) (#eq? @a.lhs \"@variable_name\"))",
-        "@abc", "",holes, HashSet::new());
+  let rule = piranha_rule! {
+      name= "test",
+      query= "(
+        ((assignment_expression left: (_) @a.lhs right: (_) @a.rhs) @abc)
+        (#eq? @a.lhs \"@variable_name\")
+      )",
+      replace_node = "abc",
+      replace = "",
+      holes = ["variable_name"]
+  };
```

Comment:
Ok, the actual macro definition is a bit clunky, but I have to admit this is **way** easier to read than the `Rule::new` example.

Any reason why using the builder directly is not a good option? Less idiomatic in Rust?

Reply:
I think builders is as idomatic as it could get. I feel it gets too verbose. But folks can always use Builder if they want to. Its available for them.